### DISCUSSION TITLE
RNSScreen in analytics events

### DISCUSSION THREAD

**Discussion by thomasblom** [invertase/react-native-firebase#7247] [Q&A]
I'm using `@react-native-firebase/analytics` in combination with `react-native-screens` . In every event there is a property `firebase_screen_class` that contains the current classname of the screen.
By default `react-native-screens` sets this property as `RNSScreen`, but i change the value with `analytics().logScreenView(...)`.
After a couple of events it changes back to `RNSScreen` and i'm not sure where that happens..

Does anyone know more about this value and where it comes from and how it resets to the initial value?

Thanks in advance!

**Comment by Panda-ref** [comment]
Happens to us as well. Mostly happens on iOS.
```
const handleNavigationStateChange = useCallback(async () => {
    const previousRouteName = routeNameRef.current;
    const currentRouteName = navigationRef.current.getCurrentRoute().name;

    if (previousRouteName !== currentRouteName) {
      await logScreenView({
        screen_name: currentRouteName,
        screen_class: currentRouteName,
      });
    }
    routeNameRef.current = currentRouteName;
  }, []);
```


Always keeping the @react-native-firebase/* packages up to date with dependabot.
@mikehardy Is there any recommendations on how to avoid that or what can cause it?

**Comment by Romaric-CH** [comment]
I have the same issue. Any updates on this ? 🙏

**Comment by ammaarkhan** [comment]
Were any of you able to find a fix for this? @thomasblom @Panda-ref @Romaric-CH @mikehardy

**Reply by ammaarkhan** [reply]
I'm trying out and seeing if stopping the automatic screen tracking might fix it. Credit: https://stackoverflow.com/questions/61056541/how-to-remove-mainactivity-and-uiviewcontroller-from-screen-tracking-in-react-na/74313385#74313385.

**Reply by tadeumx1** [reply]
This solution fix that ?

**Reply by ammaarkhan** [reply]
Oh sorry I forget about updating, yes it did!

**Reply by daviddamilola** [reply]
hi @ammaarkhan , did this solution mean that those values are set as not set in the firebase dashboard?
<img width="1011" alt="Screenshot 2024-11-10 at 14 24 30" src="https://github.com/user-attachments/assets/595c7a13-9a01-4c10-ac85-37f212e8c242">

**Reply by ammaarkhan** [reply]
hey. I'm sorry unfortunately I cannot recall exactly. I stopped working on apps early this year so I haven't been looking at it.

**Comment by ghoshabhi** [comment]
Bumping this, can someone please respond to confirm if this is still the right way to disable `RNSScreen`, `UIViewController`, `MainActivity` from appearing in Google Analytics ?
https://rnfirebase.io/analytics/usage#disable-screenview-tracking

**Comment by RustyPhillips** [comment]
I've found that in v23.4.0, this breaks:
```
logEvent(analytics, 'screen_view',  {firebase_screen: 'myScreenName', firebase_screen_class: 'myScreenClass'});
```

...but this works:
```
logScreenView(analytics, {screen_name: 'myScreenName', screen_class: 'myScreenClass'});
```

Note that both the names of the functions and parameters have changed.

**Update:** the correct approach is actually this:
```
logEvent(analytics, 'screen_view',  {screen_name: 'myScreenName', screen_class: 'myScreenClass'});
```
However, because the typescript definitions are currently borked, you'll get compile/linting errors. It looks like it will be [fixed in the next release](https://github.com/invertase/react-native-firebase/issues/8791).
